### ISSUE TITLE
YOLOE train

### ISSUE THREAD

**Issue by huangqiu15444** [ultralytics/ultralytics#21657]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

1、If I want to train YOLOE from scratch, should the file directory format of Objects365v1 follow that of the coco8 - seg dataset?
2、If I want to fine-tune my own dataset using a pre-trained model, do I need the text features of the dataset for supervised training?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @huangqiu15444, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly. An Ultralytics engineer will also review your issue and assist you soon.

We recommend a visit to the Docs for guidance where you can find many Python and CLI usage examples and where many of the most common questions may already be answered:
- Docs: https://docs.ultralytics.com/
- Python: https://docs.ultralytics.com/usage/python/
- CLI: https://docs.ultralytics.com/usage/cli/

If this is a 🐛 Bug Report, please provide a minimum reproducible example to help us debug it:
- MRE: https://docs.ultralytics.com/help/minimum-reproducible-example/

If this is a custom training ❓ Question, please provide as much information as possible, including:
- Your dataset structure and a few image/label samples
- The exact command or code you ran and full logs
- Model and dataset YAMLs
- System details (OS, Python, PyTorch, CUDA)
And verify you are following our Tips for Best Training Results:
- Training Tips: https://docs.ultralytics.com/guides/model-training-tips/

Join the Ultralytics community where it suits you best:
- Discord for real-time chat 🎧: https://discord.com/invite/ultralytics
- Discourse for in-depth discussions: https://community.ultralytics.com/
- Subreddit for community threads: https://www.reddit.com/r/Ultralytics/

## Upgrade

Upgrade to the latest ultralytics package including all requirements in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

- Requirements: https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml
- Python>=3.8: https://www.python.org/
- PyTorch>=1.8: https://pytorch.org/get-started/locally/

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See GCP Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/
- Amazon Deep Learning AMI. See AWS Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/
- Docker Image. See Docker Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/ <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all Ultralytics CI tests are currently passing. CI tests verify correct operation of all YOLO Modes and Tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help us assist with your YOLOE training questions:
- Please share the dataset directory structure you plan to use for Objects365v1 and the YAML you intend to train with
- If fine-tuning with a pretrained model, include which model you’re using and clarify whether your dataset includes any text annotations
- Provide the exact commands or scripts you plan to run and any initial errors or logs you’ve encountered

Thanks for the detailed info in advance — it will help us guide you faster 🙌

**Comment by Y-T-G** [comment]
1. Yes.
2. The class names are used as the text prompt during training

**Comment by huangqiu15444** [comment]
> 1. Yes.
> 2. The class names are used as the text prompt during training

The json files about Objects365v1 provided in the project are quite large. Is there any tool that can quickly obtain the seg-format dataset？

**Comment by Y-T-G** [comment]
Ultralytics would download it automatically. You just need to pass `Objects365.yaml`

**Comment by huangqiu15444** [comment]
Sorry to trouble you. I'm wondering if I have two datasets of the same scene. One of the datasets is annotated with 80 categories according to the COCO data format, while in the other dataset, only a few of these categories are annotated. Can these two datasets be used for training YOLOE simultaneously? Will it have any impact on the training results?

**Comment by glenn-jocher** [comment]
Yes—you can train with both, but make sure they share an identical `names` mapping (same class order/IDs) across a single combined dataset; if images in the subset dataset contain other classes that are present but unlabeled, those objects become background and will hurt those classes, so either relabel those images or exclude them/limit that dataset to images where only its annotated classes appear; see the YOLOE training guidance for data setup and fine-tuning in the docs at [YOLOE: Training and fine-tuning](https://docs.ultralytics.com/models/yoloe/).
